core/domain/suggestion_domain.py

```diff
@@ -72,3 +66,196 @@ def to_dict(self):
+        self.final_reviewer_id = final_reviewer_id
+        self.change_cmd = change_cmd
+        self.score_category = score_category
+
+    def validate(self):
+        """Validates a suggestion object of type
+        SUGGESTION_TYPE_EDIT_STATE_CONTENT.
+
+        Returns:
+            bool. The validity of the suggestion object.
+        """
+        super(SuggestionEditStateContent, self).validate()
+
+        if 'cmd' not in self.change_cmd:
+            raise utils.ValidationError(
+                'Expected change_cmd to contain a cmd key')
+
+        if self.change_cmd['cmd'] != exp_domain.CMD_EDIT_STATE_PROPERTY:
+            raise utils.ValidationError(
+                'Expected cmd to be %s, recieved %s' % (
+                    exp_domain.CMD_EDIT_STATE_PROPERTY, self.change_cmd['cmd']))
+
+        if 'property_name' not in self.change_cmd:
+            raise utils.ValidationError(
+                'Expected change_cmd to contain a property_name key')
+
+        if (
+                self.change_cmd['property_name'] !=
+                exp_domain.STATE_PROPERTY_CONTENT):
+            raise utils.ValidationError(
+                'Expected property_name to be %s, recieved %s' % (
+                    exp_domain.STATE_PROPERTY_CONTENT,
+                    self.change_cmd['property_name']))
+
+        if 'state_name' not in self.change_cmd:
+            raise utils.ValidationError(
+                'Expected change_cmd to contain a state_name key')
+
+        states = exp_services.get_exploration_by_id(self.target_id).states
+        if self.change_cmd['state_name'] not in states:
+            raise utils.ValidationError(
+                'Expected %s to be a valid state name' %
+                self.change_cmd['state_name'])
+
+    def accept(self, commit_message):
+        """Accepts the suggestion.
+
+        Args:
+            commit_message: str. The commit message.
+        """
+        change_list = [self.change_cmd]
+        exp_services.update_exploration(
+            self.final_reviewer_id, self.target_type, change_list,
+            commit_message, is_suggestion=True)
+
+    @classmethod
+    def from_dict(cls, suggestion_dict):
+        """Return a Suggestion object of type from a dict.
+
+        Args:
+            suggestion_dict: dict. The dict representation of the suggestion.
+
+        Returns:
+            BaseSuggestion. The corresponding Suggestion domain object.
+        """
+        suggestion = cls(
+            suggestion_dict['suggestion_id'],
+            suggestion_dict['target_id'],
+            suggestion_dict['target_version_at_submission'],
+            suggestion_dict['status'], suggestion_dict['author_id'],
+            suggestion_dict['assigned_reviewer_id'],
+            suggestion_dict['final_reviewer_id'], suggestion_dict['change_cmd'],
+            suggestion_dict['score_category'])
+
+        return suggestion
+
+
+SUGGESTION_TYPE_TO_DOMAIN_CLASS = {
```

Comment:
Minor renaming, sorry: SUGGESTION_TYPES_TO_DOMAIN_CLASSES (since eventually we'll have more)

Reply:
Done